core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java

```diff
@@ -154,4 +156,48 @@ public static boolean isDSWDatasource( Domain domain ) {
     }
     return !isMetadataDatasource( domain );
   }
+
+  protected String isMetaDataSource( String id ) {
+    Domain domain;
+    try {
+      domain = metadataDomainRepository.getDomain( id );
+      if ( domain == null ) {
+        return null;
+      }
+    } catch ( Exception e ) { // If we can't load the domain then we MUST return false
+      return null;
+    }
+    return isMetadataDatasource( domain ) ? id : null;
+  }
+
+  protected int noOfThreads() {
+    IPluginResourceLoader resLoader = PentahoSystem.get( IPluginResourceLoader.class, null );
+    String noOfThreadsAsString = null;
+    int noOfThreads = 1;
+    try {
+      noOfThreadsAsString =
```

Comment:
Are they any unit tests either directly or indirectly that verify the logic for `DatasourceService#noOfThreads`?